src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -708,10 +782,80 @@ def _bundle(self):
         )
         compiler = NativeAppCompiler(bundle_context)
         compiler.compile_artifacts()
+
+        if self._entity_model.children:
+            # Bundle children and append their SQL to setup script
+            # TODO Consider re-writing the logic below as a processor
+            children_sql = self._bundle_children(action_ctx=action_ctx)
+            setup_file_path = find_setup_script_file(deploy_root=self.deploy_root)
+            with open(setup_file_path, "r", encoding="utf-8") as file:
+                existing_setup_script = file.read()
+            if setup_file_path.is_symlink():
+                setup_file_path.unlink()
+            with open(setup_file_path, "w", encoding="utf-8") as file:
```

Comment:
I think we should use SecurePath here.

Reply:
Yes, that's a good suggestion. I'll add it as part of a future PR since it's an existing logic in `find_setup_script_file()` and I want to contain all new logic behind the feature flag.